### ISSUE TITLE
`object-curly-spacing`: false-positive with block comments

### ISSUE THREAD

**Issue by silverwind** [eslint-stylistic/eslint-stylistic#1070]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

A block comment in the same line as the opening curly bracket of an object literal or type alias triggers a false-positive. The same does not happen with line comments.

```ts
const x = {
  foo: { /* false positive */
    bar: "",
  }
};
```

### Reproduction

https://stackblitz.com/edit/github-72jmgcmx?file=src%2Findex.ts

### System Info

```shell
any
```

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests
